OBAApplicationDelegate.m

```diff
@@ -224,6 +236,55 @@ - (void)applicationWillResignActive:(UIApplication *)application {
     self.active = NO;
 }
 
+#pragma mark Shortcut Items
+
+- (void)application:(UIApplication *)application performActionForShortcutItem:(nonnull UIApplicationShortcutItem *)shortcutItem completionHandler:(nonnull void (^)(BOOL))completionHandler {
+    NSString *shortcutIdentifier = shortcutItem.type;
+
+    if ([shortcutIdentifier isEqualToString:kApplicationShortcutMap]) {
+        [self.tabBarController setSelectedViewController:self.mapNavigationController];
+
+        [self.mapNavigationController popToRootViewControllerAnimated:NO];
+        [self.mapViewController onCrossHairsButton:self];
+    } else if ([shortcutIdentifier isEqualToString:kApplicationShortcutBookmarks]) {
+        [self.tabBarController setSelectedViewController:self.bookmarksNavigationController];
+
+        [self.bookmarksNavigationController popToRootViewControllerAnimated:NO];
+    } else if ([shortcutIdentifier isEqualToString:kApplicationShortcutRecents]) {
+        [self.tabBarController setSelectedViewController:self.recentsNavigationController];
+
+        NSString *firstStopId = shortcutItem.userInfo[@"stopIds"][0];
+        if (firstStopId) {
```

Comment:
This'll crash if the @"stopIds" array exists but is empty. Why not change it to:

    if (shortcutItem.userInfo[@"stopIds"].count > 0)

and kill two birds with one stone.